utility/src/test/java/org/oppia/util/profile/DirectoryManagementUtilTest.kt

```diff
@@ -91,9 +90,70 @@ class DirectoryManagementUtilTest {
   }
 
   @Test
-  @ExperimentalCoroutinesApi
-  fun testDirectoryManagementUtil() = runBlockingTest(coroutineContext) {
-    // TODO(#16): Finish test cases with full implementation
+  fun testUtil_createDir_checkDirExists() {
+    directoryManagementUtil.getOrCreateDir(TEST_DIRECTORY_1)
+
+    val dir = File(getAbsoluteDirPath(TEST_DIRECTORY_1))
+
+    assertThat(dir.exists()).isTrue()
+  }
+
+  @Test
+  fun testUtil_createDir_getSameDir_checkDirsEquality() {
+    val dir = directoryManagementUtil.getOrCreateDir(TEST_DIRECTORY_1)
+
+    val sameDir = directoryManagementUtil.getOrCreateDir(TEST_DIRECTORY_1)
+
+    assertThat(dir).isEqualTo(sameDir)
+  }
+
+  @Test
+  fun testUtil_createDir_createDifferentDir_checkDirsNotEqual() {
+    val dir = directoryManagementUtil.getOrCreateDir(TEST_DIRECTORY_1)
+
+    val diffDir = directoryManagementUtil.getOrCreateDir(TEST_DIRECTORY_2)
+
+    assertThat(dir).isNotEqualTo(diffDir)
+  }
+
+  @Test
+  fun testUtil_createDir_restartApplication_checkDirExists() {
+    directoryManagementUtil.getOrCreateDir(TEST_DIRECTORY_1)
+
+    setUpTestApplicationComponent()
+
+    val dir = File(getAbsoluteDirPath(TEST_DIRECTORY_1))
+    assertThat(dir.exists()).isTrue()
+  }
+
+
+  @Test
+  fun testUtil_createDir_deleteDir_checkDirDoesNotExist() {
+    directoryManagementUtil.getOrCreateDir(TEST_DIRECTORY_1)
+
+    val success = directoryManagementUtil.deleteDir(TEST_DIRECTORY_1)
+
+    val dir = File(getAbsoluteDirPath(TEST_DIRECTORY_1))
+    assertThat(dir.exists()).isFalse()
+    assertThat(success).isTrue()
+  }
+
+  @Test
+  fun testUtil_createDir_createFilesUnderDir_deleteDir_checkDirDoesNotExist() {
```

Comment:
Here & elsewhere: this test name doesn't quite read right. I suggest always keeping the operation being tested next to the ``test`` word to keep things in context, e.g.:

``testDeleteDir_forCreatedDir_withContainedFiles_dirIsDeleted``

The first part is always the action, the last part is always the outcome, and everything in the middle comprise the conditions that bridge the two together.